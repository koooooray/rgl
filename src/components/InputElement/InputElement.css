.Input {
    width: 100%;
    display: flex;
    flex-wrap: wrap;
    flex-direction: row;
    margin-bottom: 15px;
    min-height: 95px;
}

.Label {
    width: 40%;
    font-weight: bold;
    margin-right: 30px;
    height: 50px;
    line-height: 50px;
    /* text-align: center; */
}

.ValidationError {
    width: 100%;
    color: red;
    padding-left: 40%;
    margin-left: 30px;
    margin-top: 5px;
}

input,
select {
    padding-left: 10px;
    font-size: large;
    height: 50px;
    background-color: #eee;
    border: 1px solid gray;
    border-radius: 3px;
}

.ToggleContainer {
    display: block;
    position: relative;
    padding-left: 35px;
    margin-top: 12px;
    cursor: pointer;
    font-size: 22px;
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
}

.ToggleContainer input {
    position: absolute;
    opacity: 0;
    cursor: pointer;
}

.ToggleCheckmark {
    position: absolute;
    top: 0;
    left: 0;
    height: 25px;
    width: 25px;
    background-color: #eee;
    border: 1px solid grey;
}

.ToggleContainer:hover input~.ToggleCheckmark {
    background-color: #ccc;
}

.ToggleContainer input:checked~.ToggleCheckmark {
    background-color: #2196F3;
}

.ToggleCheckmark:after {
    content: "";
    position: absolute;
    display: none;
}

.ToggleContainer input:checked~.ToggleCheckmark:after {
    display: block;
}

.ToggleContainer .ToggleCheckmark:after {
    left: 9px;
    top: 5px;
    width: 5px;
    height: 10px;
    border: solid white;
    border-width: 0 3px 3px 0;
    -webkit-transform: rotate(45deg);
    -ms-transform: rotate(45deg);
    transform: rotate(45deg);
}

.Invalid {
    color: red;
}