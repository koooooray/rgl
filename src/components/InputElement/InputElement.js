import React from 'react';
import classes from "./InputElement.css";

const InputElement = (props) => {
    
    let inputElement = null;
    const inputClasses = [classes.Input];

    if(props.invalid && props.touched){
        inputClasses.push(classes.Invalid);
    }

    switch(props.elementType){
        // case('input'):
        //     inputElement = <input
        //     {...props.elementConfig} />
        //     break;
        case('select'):
            inputElement = <select 
                value={props.value}
                onChange={props.changed}>
                {props.elementConfig.hasEmptyOption ? <option value={''}>-</option> : ''}
                {props.elementConfig.options.map(option=>(
                    <option 
                        key={option.value} 
                        value={option.value}>
                        {option.displayName}
                    </option>
                ))}
            </select>
            break;
        case('toggle'):
            const checked = props.elementConfig ? props.elementConfig.checked : false;
            inputElement = (<label className={classes.ToggleContainer}>
                    <input type="checkbox" onChange={props.changed} defaultChecked={checked}/>
                    <span  className={classes.ToggleCheckmark}></span>
                </label>)
            break;
        default:
            inputElement = null;
    }

    return !inputElement ? "" : (
        <div className={inputClasses.join(" ")}>
            <label className={classes.Label}>{props.label}</label>
            {inputElement}
            <div className={classes.ValidationError + ' validation'}>{props.invalid && props.touched ? props.validationMessage : ''}</div>
        </div>
    );
}

export default InputElement;