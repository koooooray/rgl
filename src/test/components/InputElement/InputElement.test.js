import React from "react";
import ReactShallowRenderer from "react-test-renderer/shallow";
import InputElement from "../../../components/InputElement/InputElement";

import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });
import { shallow } from "enzyme";

const countries = [
  {
    value: "usa",
    supportedCountry: true,
    displayName: "United States of America",
    selected: false
  },
  {
    value: "ger",
    supportedCountry: true,
    displayName: "Germany",
    selected: false
  },
  {
    value: "fra",
    supportedCountry: false,
    displayName: "France",
    selected: false
  },
  {
    value: "uk",
    supportedCountry: true,
    displayName: "United Kingdom",
    selected: false
  },
  {
    value: "hol",
    supportedCountry: false,
    displayName: "Netherlands",
    selected: false
  },
  {
    value: "can",
    supportedCountry: false,
    displayName: "Canada",
    selected: false
  }
];
const elementConfig = {
  options: countries,
  hasEmptyOption: false
};

test("should skip not properly configured inputs", () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<InputElement />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test("should render correctly configured inputs", () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(
    <InputElement
      key={"some-field"}
      elementType={"some-type"}
      label={"some-field"}
      value={false}
      invalid={false}
      validationMessage={"some-validation-message"}
      touched={false}
      changed={event => true}
    />
  );
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test("type select renders all options", () => {
  const wrapper = shallow(
    <InputElement
      key={"some-field"}
      elementType={"select"}
      elementConfig={elementConfig}
      label={"some-field"}
      value={false}
      invalid={false}
      validationMessage={"some-validation-message"}
      touched={false}
      changed={event => true}
    />
  );
  expect(wrapper.find("select").length).toBe(1);
  expect(wrapper.find("option").length).toBe(6);
});

test("type select adds empty option", () => {
  const config = { ...elementConfig, hasEmptyOption: true };
  const wrapper = shallow(
    <InputElement
      key={"some-field"}
      elementType={"select"}
      elementConfig={config}
      label={"some-field"}
      value={false}
      invalid={false}
      validationMessage={"some-validation-message"}
      touched={false}
      changed={event => true}
    />
  );
  expect(wrapper.find("select").length).toBe(1);
  expect(wrapper.find("option").length).toBe(7);
});

test("type toggle renders only one input", () => {
  const wrapper = shallow(
    <InputElement
      key={"some-field"}
      elementType={"toggle"}
      elementConfig={{}}
      label={"some-field"}
      value={true}
      invalid={false}
      validationMessage={"some-validation-message"}
      touched={false}
      changed={event => true}
    />
  );
  expect(wrapper.find("input").length).toBe(1);
});

