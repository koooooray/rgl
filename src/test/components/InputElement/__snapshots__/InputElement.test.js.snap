// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly configured inputs 1`] = `""`;

exports[`should skip not properly configured inputs 1`] = `""`;
